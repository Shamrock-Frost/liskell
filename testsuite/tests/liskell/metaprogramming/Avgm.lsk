(defmodule Avgm _ (Liskell LskPrelude))

(defenv (lambda ((LskEnv e p t d))
	  (return (LskEnv (backquote-dspr trf_expr e) p t (defmacro-dspr trf_decl (def-hdspr-dsprs trf_decl d))))))

;; Naive average
;; (define (avg lst)
;;   (/ (sum lst) (length lst)))
;; Improved avg macro, compile-time list traversal

(defmacro (avgm pts)
  `(/ (sum ([] ,@pts))
    ,(length pts)))
