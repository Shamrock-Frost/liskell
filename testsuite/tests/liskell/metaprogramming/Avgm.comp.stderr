
Avgm.lsk:4:20: Not in scope: `backquote-dspr'

Avgm.lsk:4:76: Not in scope: `def-hdspr-dsprs'
Loading package base ... linking ... done.
Loading package template-haskell ... linking ... done.
Loading package readline-1.0 ... linking ... done.
Loading package unix-1.0 ... linking ... done.
Loading package Cabal-1.1.7 ... linking ... done.
Loading package regex-base-0.71 ... linking ... done.
Loading package regex-posix-0.71 ... linking ... done.
Loading package regex-compat-0.71 ... linking ... done.
Loading package haskell98 ... linking ... done.
Loading package ghc-6.7 ... linking ... done.
